### PR TITLE
01990: Use correct dedicated types of assertions

### LINKED ISSUE #1990: Use correct dedicated types of assertions
https://sonarcloud.io/project/issues?id=com.hedera.hashgraph%3Ahedera-services&resolved=false&rules=java%3AS5785&types=CODE_SMELL

### LINKED ISSUE #2115: Remove unused AccountBalance class and its unit test suite

### PR DESCRIPTION
**Pull request by qnswirlds**
Close #1990
Close #2115

### CODE REVIEW

File: hapi-utils/src/test/java/com/hedera/services/throttles/DeterministicThrottleTest.java [hunk 1]
```diff
@@ -302,8 +302,9 @@ void equalsWorks() {
 		final var throttle1 = DeterministicThrottle.withTpsAndBurstPeriodNamed(tps, 2, name);
 		final var bucketThrottle = BucketThrottle.withTps(tps);
 
-		assertTrue(throttle.equals(throttle1));
-		assertFalse(throttle.equals(null));
-		assertFalse(throttle.equals(bucketThrottle));
+		assertEquals(throttle, throttle1);
+		final var equalsForcedCallResult = throttle.equals(null);
+		assertFalse(equalsForcedCallResult);
```
Comment (anighanta) [inline hapi-utils/src/test/java/com/hedera/services/throttles/DeterministicThrottleTest.java:307]: `assertNotNull(throttle);` would be easier to understand

File: hedera-node/src/test/java/com/hedera/services/context/domain/topic/MerkleTopicTest.java [hunk 2]
```diff
@@ -131,24 +133,27 @@ void copyConstructorWithValuesTestingAccessors() throws Exception {
...
 	void equalsDefault() {
-		assertTrue(new MerkleTopic().equals(new MerkleTopic()));
+		assertEquals(new MerkleTopic(), new MerkleTopic());
 	}
 
 	@Test
 	void equalsNull() {
-		assertFalse(new MerkleTopic().equals(null));
+		final var equalsForcedCallResult = new MerkleTopic().equals(null);
+		assertFalse(equalsForcedCallResult);
```
Comment (anighanta) [inline hedera-node/src/test/java/com/hedera/services/context/domain/topic/MerkleTopicTest.java:142]: `assertNotNull(new MerkleTopic());`

File: hedera-node/src/test/java/com/hedera/services/records/EarliestEntryExpiryTest.java [hunk 3]
```diff
@@ -25,37 +25,34 @@
...
-		assertTrue(ere.equals(new EarliestRecordExpiry(5L, asAccount("0.0.5"))));
-		assertFalse(ere.equals(new EarliestRecordExpiry(6L, asAccount("0.0.5"))));
+		final var ere = new EarliestRecordExpiry(5L, asAccount("0.0.5"));
+
+		assertEquals(ere, ere);
+		assertNotEquals(ere, new Object());
+		assertEquals(ere, new EarliestRecordExpiry(5L, asAccount("0.0.5")));
+		assertNotEquals(ere, new EarliestRecordExpiry(6L, asAccount("0.0.5")));
+		assertNotEquals(ere, new EarliestRecordExpiry(5L, asAccount("0.0.6")));
+		final var equalsForcedCallResult = ere.equals(null);
```
Comment (anighanta) [inline hedera-node/src/test/java/com/hedera/services/records/EarliestEntryExpiryTest.java:40]: Can simplify using `assertNotNull` here as well.

File: hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleEntityIdTest.java [hunk 4]
```diff
@@ -23,126 +23,107 @@
...
 
-		// when:
 		three.setShard(1);
 		three.setRealm(2);
 		three.setNum(3);
 
-		// then:
-		assertNotEquals(null, one);
+		final var equalsForcedCallResult = one.equals(null);
+		assertFalse(equalsForcedCallResult);
```
Comment (anighanta) [inline hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleEntityIdTest.java:69]: Can simplify using assertNotNull here as well.

File: hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleOptionalBlobTest.java [hunk 5]
```diff
@@ -159,147 +149,117 @@ void merkleMethodsWork() {
...
-		// then:
-		assertNotEquals(null, one);
-		assertNotEquals(four, one);
+		final var one = new MerkleOptionalBlob();
+		final var two = new MerkleOptionalBlob(stuff);
+		final var three = new MerkleOptionalBlob(stuff);
+		final var twoRef = two;
+
+		final var equalsForcedCallResult = one.equals(null);
+		assertFalse(equalsForcedCallResult);
```
Comment (anighanta) [inline hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleOptionalBlobTest.java:257]: Can simplify using assertNotNull here as well.

**Review by anighanta — APPROVED** [review]
LGTM. see incline comments

File: hapi-utils/src/test/java/com/hedera/services/throttles/DeterministicThrottleTest.java (hunk 1 above)
Comment (qnswirlds) [reply hapi-utils/src/test/java/com/hedera/services/throttles/DeterministicThrottleTest.java:307]: @anighanta, but they don't assert/run the same code. Please read my variable name!

**Pull request merged**